PICMI_Python/applied_fields.py

```diff
@@ -54,35 +58,16 @@ class PICMI_AnalyticAppliedField(_ClassWithInit):
       - lower_bound=[None,None,None]: Lower bound of the region where the field is applied (vector) [m]
       - upper_bound=[None,None,None]: Upper bound of the region where the field is applied (vector) [m]
     """
-    def __init__(self, Ex_expression=None, Ey_expression=None, Ez_expression=None,
-                       Bx_expression=None, By_expression=None, Bz_expression=None,
-                 lower_bound=[None,None,None], upper_bound=[None,None,None],
-                 **kw):
-
-        self.Ex_expression = Ex_expression
-        self.Ey_expression = Ey_expression
-        self.Ez_expression = Ez_expression
-        self.Bx_expression = Bx_expression
-        self.By_expression = By_expression
-        self.Bz_expression = Bz_expression
-
-        self.lower_bound = lower_bound
-        self.upper_bound = upper_bound
-
-        # --- Find any user defined keywords in the kw dictionary.
-        # --- Save them and delete them from kw.
-        # --- It's up to the code to make sure that all parameters
-        # --- used in the expression are defined.
-        self.user_defined_kw = {}
-        for k in list(kw.keys()):
-            if ((self.Ex_expression is not None and re.search(r'\b%s\b'%k, self.Ex_expression)) or
-                (self.Ey_expression is not None and re.search(r'\b%s\b'%k, self.Ey_expression)) or
-                (self.Ez_expression is not None and re.search(r'\b%s\b'%k, self.Ez_expression)) or
-                (self.Bx_expression is not None and re.search(r'\b%s\b'%k, self.Bx_expression)) or
-                (self.By_expression is not None and re.search(r'\b%s\b'%k, self.By_expression)) or
-                (self.Bz_expression is not None and re.search(r'\b%s\b'%k, self.Bz_expression))):
-                self.user_defined_kw[k] = kw[k]
-                del kw[k]
+    @autoargs(exclude=['kw'])
+    def __init__(self, Ex_expression : Expression = None,
+                       Ey_expression : Expression = None,
+                       Ez_expression : Expression = None,
+                       Bx_expression : Expression = None,
+                       By_expression : Expression = None,
+                       Bz_expression : Expression = None,
+                       lower_bound : VectorFloat3 = [None,None,None],
```

Comment:
Is assigning a list as default argument value (old and new) actually possible?

Common Python bug:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Yes, this code works as is, and I am aware of the issue. In this case, however, it is not expected that the argument, `lower_bound`, will be modified, but only read. An alternative here would be to have `lower_bound` default to `None`. This would be more consistent with other vectors that default to `None`. In this PR, I wanted to avoid changing the interface, but the change can be made.